src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/AddContextDataToShardedReads.java

```diff
@@ -0,0 +1,35 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.dataflow.sdk.values.KV;
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+
+/**
+ *
+ */
+public class AddContextDataToShardedReads {
+
+    public static ReadContextData AddContextDataToShardedReads(KV<String,Iterable<GATKRead>> shardedReads, RefAPISource refSource, PipelineOptions opts, SAMSequenceDictionary refDictionary) {
+        SimpleInterval shard = new SimpleInterval(shardedReads.getKey());
+        // how many extra bases to grab on either side.
+        // make sure we're not going past the end of the contig.
+        int margin = 100;
+        // 1-based
+        int start = shard.getStart()-margin;
+        if (start<1) start=1;
+        final int refContigLength = refDictionary.getSequence(shard.getContig()).getSequenceLength();
+        // included
+        int end = shard.getEnd()+margin;
+        if (end>refContigLength) end = refContigLength;
+
+
+        SimpleInterval paddedShard = new SimpleInterval(shard.getContig(), start, end);
```

Comment:
Extract static utility method `padInterval(interval, padding, contigLength)` to `IntervalUtils`, and call it here